cmd/cosign/cli/init.go

```diff
@@ -18,17 +18,44 @@ package cli
 import (
 	"context"
 	"flag"
+	"io/ioutil"
+	"net/http"
+	"path/filepath"
+	"strings"
 
 	"github.com/peterbourgon/ff/v3/ffcli"
 	ctuf "github.com/sigstore/cosign/pkg/cosign/tuf"
 )
 
+func loadFileOrURL(fileRef string) ([]byte, error) {
+	var raw []byte
+	var err error
+	if strings.HasPrefix(fileRef, "http://") || strings.HasPrefix(fileRef, "https://") {
+		// #nosec G107
+		resp, err := http.Get(fileRef)
+		if err != nil {
+			return nil, err
+		}
+		defer resp.Body.Close()
+		raw, err = ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		raw, err = ioutil.ReadFile(filepath.Clean(fileRef))
+		if err != nil {
+			return nil, err
+		}
+	}
+	return raw, nil
+}
+
 func Init() *ffcli.Command {
 	var (
 		flagset = flag.NewFlagSet("cosign init", flag.ExitOnError)
 		// TODO: Support HTTP mirrors as well
 		mirror    = flagset.String("mirror", "sigstore-tuf-root", "GCS bucket to a SigStore TUF repository.")
-		root      = flagset.String("root", ".sigstore/keys.json", "path to trusted initial root.")
+		root      = flagset.String("root", "https://raw.githubusercontent.com/sigstore/root-signing/main/ceremony/2021-06-18/repository/1.root.json", "path to trusted initial root.")
```

Comment:
wdyt about somehow sticking this into a GCR or GCS bucket, and hardcoding the hash of the resulting file into cosign somehow? This is the root root root, so the hash should never change.

Reply:
ooh thanks! i like sticking the hash in. 
 (and actually i think there's http fetches for buckets too? that might make it comptaible with a url flag)

Reply:
Yea! "storage.googleapis.com/<bucket>/<object>" should work.

Reply:
is it really necessary to stick the sha in if it's validated in the snapshot.json?